### PR TITLE
Add masking policy resource

### PR DESCRIPTION
**Pull request by ajbosco**
## Test Plan

* [x] acceptance tests

* [x] unit tests

## References

*

### CODE REVIEW

File: docs/resources/masking_policy.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+- **return_data_type** (String, Required) Specifies the data type to return.
+- **schema** (String, Required) The schema in which to create the masking policy.
+- **value_data_type** (String, Required) Specifies the data type to mask.
+
+### Optional
+
+- **comment** (String, Optional) Specifies a comment for the masking policy.
+- **id** (String, Optional) The ID of this resource.
+
+
```
Comment (alldoami) [inline docs/resources/masking_policy.md:30]: Can you put in an example and how to import this resource?

Comment (ajbosco) [reply docs/resources/masking_policy.md:30]: 👍 Added

File: docs/resources/role_grants.md [hunk 2]
```diff
@@ -32,6 +32,8 @@ resource "snowflake_role" "other_role" {
 }
 
 resource "snowflake_role_grants" "grants" {
```
Comment (alldoami) [inline docs/resources/role_grants.md:34]: This change was recently merged... https://github.com/chanzuckerberg/terraform-provider-snowflake/pull/383/files#diff-cafd3d35c9890667669bb5ce2863de6e2965954c8c8a345065a829933780d28aR34

**Comment by alldoami** [comment]
Looks like the acceptance test failed: https://github.com/chanzuckerberg/terraform-provider-snowflake/runs/1665872992?check_suite_focus=true

**Review by alldoami — CHANGES_REQUESTED** [review]
Looks like the acceptance tests are failing

**Comment by ajbosco** [comment]
> Looks like the acceptance tests are failing

I think they should be passing now.

**Comment by ajbosco** [comment]
Looks like the id generation changes reverted so some changes need to happen here unless it is going to be merged again. https://github.com/chanzuckerberg/terraform-provider-snowflake/pull/407

**Comment by alldoami** [comment]
> Looks like the id generation changes reverted so some changes need to happen here unless it is going to be merged again. #407

do you want to create another PR to add what you need back? 🙏

**Review by alldoami — APPROVED** [review]

**Comment by alldoami** [comment]
Thank you @ajbosco! 🥳

**Pull request merged**